### PR TITLE
chore: Add virtual envs

### PR DESCRIPTION
**Pull request by LazyAfternoons**
### Short description

This PR adds virtual environments support for `node` and `ruby`.

#### List of Changes
- Adds `.node-version`; 
- Adds `.ruby-version`;
- Adds `.xcode.env.local` to `.gitignore`;
- Updates `README.md` with instructions on how to run the example app.

### CODE REVIEW

File: example/.node-version
Comment (hevelius) [inline example/.node-version:1]: what do you think to add this to the root of project directory?

File: example/README.md [hunk 1]
```diff
@@ -2,78 +2,50 @@ This is a new [**React Native**](https://reactnative.dev) project, bootstrapped
...
-# using npm
-npm run ios
+# Android
+$ yarn react-native run-android
 
-# OR using Yarn
-yarn ios
+# iOS
+$ yarn react-native run-ios
 ```
```
Comment (hevelius) [inline example/README.md]: What do you think to use 
`yarn ios` or `yarn android` from the example directory?

File: example/README.md
Comment (hevelius) [inline example/README.md:1]: Is the step that runs metro server missing?

**Review by hevelius — CHANGES_REQUESTED** [review]

File: example/README.md (hunk 1 above)
Comment (LazyAfternoons) [reply example/README.md]: Done, thank you.

File: example/README.md
Comment (LazyAfternoons) [reply example/README.md:1]: There's no need to start the metro server manually `yarn react-native run-android` (or iOS) automatically starts metro if there's not an instance already running.

File: example/.node-version
Comment (LazyAfternoons) [reply example/.node-version:1]: I replaced the existing `.nvmrc` with `.node-version` in the root directory. I also downgraded it to `18.19.1` which is what's specified in `nvmrc`.

File: example/README.md
Comment (hevelius) [reply example/README.md:1]: Yes, my mistake. I confused the README of the root project. Thank you.

File: example/README.md [hunk 2]
```diff
@@ -2,78 +2,50 @@ This is a new [**React Native**](https://reactnative.dev) project, bootstrapped
...
-npm run android
+# Install dependencies
+# Run this only during the first setup and when JS dependencies change
+$ yarn install
 
-# OR using Yarn
-yarn android
+# Install podfiles when targeting iOS (ignore this step for Android)
+# Run this only during the first setup and when Pods dependencies change
+$ cd iOS && bundle exec pod install && cd ..
```
Comment (hevelius) [inline example/README.md]:
```suggestion
$ cd ios && bundle exec pod install && cd ..
```

**Review by hevelius — CHANGES_REQUESTED** [review]

**Review by hevelius — APPROVED** [review]
LGTM

**Pull request merged**